idaes/models/properties/modular_properties/base/generic_property.py

```diff
@@ -3718,32 +3821,95 @@ def _therm_cond_phase(self):
         try:
 
             def rule_therm_cond_phase(b, p):
-                return get_phase_method(b, "therm_cond_phase", p)(b, p)
+                try:
+                    return get_phase_method(b, "therm_cond_phase", p)(b, p)
+                except GenericPropertyPackageError:
+                    # Handle case where thermal conductivity isn't defined for a phase
+                    return Expression.Skip
 
             self.therm_cond_phase = Expression(
                 self.phase_list,
-                doc="Thermal conductivity for each phase",
                 rule=rule_therm_cond_phase,
             )
         except AttributeError:
             self.del_component(self.therm_cond_phase)
             raise
 
+    def _therm_cond_phase_comp(self):
+        try:
+
+            def rule_therm_cond_phase_comp(b, p, j):
+                cobj = b.params.get_component(j)
+                if (
+                    cobj.config.therm_cond_phase_comp is not None
+                    and p in cobj.config.therm_cond_phase_comp
+                    and cobj.config.therm_cond_phase_comp[p] is not None
+                ):
+                    return cobj.config.therm_cond_phase_comp[
+                        p
+                    ].therm_cond_phase_comp.return_expression(b, cobj, p, b.temperature)
+                else:
+                    # Handle case where thermal conductivity isn't defined for a particular phase-component pair
+                    return Expression.Skip
+
+            self.therm_cond_phase_comp = Expression(
+                self.phase_list,
```

Comment:
You should use the `phase_component_set` here - this will automatically exclude any invalid phase-component pairs (i.e., any component that was declared to not exist within a given phase).